### PR TITLE
Support flat config and upgrade eslint to v9

### PR DESCRIPTION
**Pull request by gracepark**
- closes https://github.com/github/web-systems/issues/2568, https://github.com/github/web-systems/issues/2569, https://github.com/github/eslint-plugin-github/issues/513

This allows us to support flat config and provides testing for both:
- Upgrades the plugin to v9
- [Flat config](https://github.com/github/eslint-plugin-github/blob/2c0f2cf9eee2f25e0c3eb2a875d93923d697f4c5/lib/configs/flat/browser.js)
- Legacy config remains in the same location
- I've created [`test-examples`](https://github.com/github/eslint-plugin-github/tree/2c0f2cf9eee2f25e0c3eb2a875d93923d697f4c5/test-examples) which allows testing for both legacy and flat configs:
  - For legacy, run: `npm run test-examples:legacy` on 8.57 without flat config
  - For flat, run: `npm run test-examples:flat` on 8.57 with flat config

## Testing
I used [`yalc`](https://www.npmjs.com/package/yalc/v/1.0.0-pre.23) to test this package with other repos:

- [clipboard-copy-element](https://github.com/github/clipboard-copy-element): This is on v8.42.0 and on legacy eslintrc.json config. I was able to "update" the `eslint-plugin-github` plugin and ran `npm run lint` and it ran as expected. 
- [codeql-action](https://github.com/github/codeql-action): eslint is on 8.57.1 and on flat config. I was able to "update" the `eslint-plugin-github`, ran `npm run lint` and it ran as expected.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -28,6 +30,31 @@ JSON ESLint config example:
...
+Import the `eslint-plugin-github`, and extend any of the configurations using `getFlatConfigs()` as needed like so:
+
+```js
+import github from 'eslint-plugin-github'
+
+export default [
+  github.getFlatConfigs().browser,
+  github.getFlatConfigs().recommended,
+  github.getFlatConfigs().react,
+  ...github.getFlatConfigs().typescript,
```
Comment (manuelpuyol) [inline README.md:44]: why is this one different than the reset?

Comment (gracepark) [reply README.md:44]: This was due to the way their recommended `tseslint.config` works as a part of [their upgraded `typescript-eslint`](https://typescript-eslint.io/blog/announcing-typescript-eslint-v7/#switching-to-typescript-eslint):

https://github.com/github/eslint-plugin-github/pull/571/files#diff-a9abd1556c7d3b1217538365a174c9d48cf9a18420eb7d92240409904d042ff0R6

It outputs an array while the others outputs objects (example, I shortened the number of rules for ease of reading):

Internal config:

```
{
  plugins: { github: { meta: [Object], rules: [Object] } },
  rules: {
    'github/authenticity-token': 'error',
    'github/js-class-name': 'error',
    'github/no-d-none': 'error'
  }
}
```

Typescript config:

```
[
  {
    rules: {
      'constructor-super': 'error',
      'for-direction': 'error',
    }
  },
  {
    name: 'typescript-eslint/base',
    languageOptions: { parser: [Object], sourceType: 'module' },
    plugins: { '@typescript-eslint': [Object] }
  },
  {
    files: [ '**/*.ts', '**/*.tsx', '**/*.mts', '**/*.cts' ],
    rules: {
      'constructor-super': 'off',
      'getter-return': 'off',
    },
    name: 'typescript-eslint/eslint-recommended'
  },
  {
    name: 'typescript-eslint/recommended',
    rules: {
      '@typescript-eslint/ban-ts-comment': 'error',
      'no-array-constructor': 'off',
    }
  },
  {
    languageOptions: { parser: [Object] },
    plugins: {
      '@typescript-eslint': [Object],
      escompatPlugin: [Object],
      github: [Object]
    },
    rules: {
      camelcase: 'off',
      'no-unused-vars': 'off',
    }
  }
]
```
So I use the spread operator to expand the elements in the typescript config array.

Comment (manuelpuyol) [reply README.md:44]: ahh makes sense, thank you

**Review by manuelpuyol — APPROVED** [review]
This looks great! I love the new test examples and that you tested different repos 😄

**Pull request merged**